subsys/net/lib/dhcpv4/dhcpv4.c

```diff
@@ -810,6 +818,45 @@ static bool dhcpv4_parse_options(struct net_pkt *pkt,
 			break;
 		}
 #endif
+#if defined(CONFIG_LOG_BACKEND_NET_USE_DHCPV4_OPTION)
+		case DHCPV4_OPTIONS_LOG_SERVER: {
+			struct sockaddr_in log_server;
+
+			/* Log server option may present 1 or more
+			 * addresses. Each 4 bytes in length. Log
+			 * servers should be listed in order
+			 * of preference.  Hence we choose the first
+			 * and skip the rest.
+			 */
+			if (length % 4 != 0U) {
+				NET_ERR("options_log_server, bad length");
+				return false;
+			}
+
+			(void)memset(&log_server, 0, sizeof(log_server));
+
+			if (net_pkt_read(pkt, log_server.sin_addr.s4_addr, 4) ||
```

Comment:
This is perfectly valid code but there is some Misra rule that says we should always check the value explicitly unless the value is boolean. So here it would be better to write
```
if (net_pkt_read(pkt, log_server.sin_addr.s4_addr, 4) < 0 ||
    net_pkt_skip(pkt, length - 4U) < 0) {
```
Currently there is no checker for these but we can be proactive to avoid future warnings.